filesystem/iso9660.ksy

```diff
@@ -5,269 +5,607 @@ meta:
   xref:
     wikidata: Q815645
   license: CC0-1.0
+  endian: be
 doc: |
   ISO9660 is standard filesystem used on read-only optical discs
   (mostly CD-ROM). The standard was based on earlier High Sierra
   Format (HSF), proposed for CD-ROMs in 1985, and, after several
   revisions, it was accepted as ISO9960:1998.
-
   The format emphasizes portability (thus having pretty minimal
   features and very conservative file names standards) and sequential
   access (which favors disc devices with relatively slow rotation
   speed).
+doc-ref: |
+  ecma-119 http://www.ecma-international.org/publications/files/ECMA-ST/Ecma-119.pdf
+  susp http://www.ymi.com/ymi/sites/default/files/pdf/Systems%20Use%20P1281.pdf
+  rrip http://www.ymi.com/ymi/sites/default/files/pdf/Rockridge.pdf
+  rras http://www.estamos.de/makecd/Rock_Ridge_Amiga_Specific
+  rrzf https://dev.lovelyhq.com/libburnia/web/wikis/Zisofs
+enums:
+  descriptor_type:
+    0x00: boot_record_volume_descriptor
+    0x01: primary_volume_descriptor
+    0x02: supplementary_volume_descriptor
+    0x03: volume_partition_descriptor
+    0xff: volume_descriptor_set_terminator
+  su_signature:
+    0x4153: rras_amiga_specific # AS
+    0x4345: susp_continuation_area # CE
+    0x434c: rrip_child_link # CL
+    0x4552: susp_extensions_reference # ER
+    0x4553: susp_extension_selector # ES
+    0x4e4d: rrip_alternate_name # NM
+    0x5044: susp_padding_field # PD
+    0x504c: rrip_parent_link # PL
+    0x504e: rrip_posix_device_number # PN
+    0x5058: rrip_posix_file_attributes # PX
+    0x5245: rrip_relocated_directory # RE
+    0x5252: rrip_extensions_in_use_indicator # RR
+    0x5346: rrip_sparse_file # SF
+    0x534c: rrip_symbolic_link # SL
+    0x5350: susp_indicator # SP
+    0x5354: susp_terminator # ST
+    0x5446: rrip_time_file # TF
+    0x5a46: rrzf_zisofs # ZF
 types:
-  vol_desc:
+  u2bi:
+    doc-ref: ecma-119 7.2.3
+    seq:
+      - id: le
+        type: u2le
+      - id: be
+        type: u2be
+  u4bi:
+    doc-ref: ecma-119 7.3.3
+    seq:
+      - id: le
+        type: u4le
+      - id: be
+        type: u4be
+  volume_descriptor:
+    doc-ref: ecma-119 8.1
     seq:
-      - id: type
+      - id: descriptor_type
+        doc-ref: ecma-119 8.1.1
         type: u1
+        enum: descriptor_type
       - id: magic
-        contents: "CD001"
+        doc-ref: ecma-119 8.1.2
+        contents: [0x43, 0x44, 0x30, 0x30, 0x31]
```

Comment:
What was the problem with a string signature?

Reply:
Well, it was 3 fields and the first was 0x00, 0x01, 0x02, 0x03 or 0xff
This I found more in line of that
Also... 100% rewrite, I did not look that much at the original in the beginning

Reply:
I guess the string was better.